Focused-Launch: Fix styles for buttons on the Plan Details View

StefanNieuwenhuis opened the pull request:
The current implementation does not match the design so we updated the disabled & hover state of unselected buttons to make it match again.

#### Testing instructions
1. Create a new site via `/start` and pick a free plan
2. Go to the block editor, append `?flags=create/focused-launch-flow` to the URL, and refresh the page
3. Click on the "Launch" button in the editor's header
4. Pick a premium domain
5. Click on "View all plans" to show the Plans grid
6. Open Dev Tools and check if:
   - [x] The disabled "Free Plan" button's text color is not changed when hovered;
   - [x] The disabled "Free Plan" button has an opacity of `0.5` when disabled;
   - [x] The disabled "Free Plan" button's text color is `gray 90`;
   - [x] The disabled "Free Plan" button's text has a strikethrough effect;
   - [x] The disabled and unselected buttons have a `1px` wide border
   - [x] The unselected buttons don't change text color when hovered

Note: copy and styles for the "Free Plan" card are spec'd separately in #47788 and solved in #47796

#### Context
Remember that the Plans Grid package is used in different places — namely:

  * New Onboarding
  * Focused Launch
  * Step by Step launch

**Please test in every environment!**

#### Expected result:

When user can select Free Plan:

![Screenshot 2020-11-25 at 13 37 59](https://user-images.githubusercontent.com/1083581/100228745-72201c80-2f23-11eb-8873-a02c1defe72b.png)

When Free plan is not selectable (because user has picked a premium domain)

<img width="1355" alt="Screenshot 2020-11-26 at 12 13 48" src="https://user-images.githubusercontent.com/1083581/100344386-04d2c100-2fe1-11eb-9f6e-74c9841612d2.png">


Fixes #47740

StefanNieuwenhuis committed:
Focused-Launch: Fix styles for buttons on the Plan Details View

The current implementation does not match the design so we updated the disabled & hover state of unselected buttons to make it match again.

ciampo commented:
Let's wait until we have an answer to https://github.com/Automattic/wp-calypso/issues/47740#issuecomment-734225642

ciampo commented:
Specs in #47740 have been slightly updated, although I don't think that the Acceptance Criteria have changed. I'm unblocking this PR, and mark it ready for review again

tjcafferkey commented:
@StefanNieuwenhuis the changes look good to me, I've tested Step by Step launch and Gutenboarding too but they look fine as far as I can tell.

It also looks like they use the Plans Accordion rather than the Plans Grid so this change shouldn't have any visual changes for them, is that right?

StefanNieuwenhuis commented:
> It also looks like they use the Plans Accordion rather than the Plans Grid so this change shouldn't have any visual changes for them, is that right?

I think that's correct. @ciampo can you verify this please?

ciampo commented:
Hey @StefanNieuwenhuis , just noticed a missing spec (which I have now added to the list:

> The disabled "Free Plan" button has an opacity of 0.5 when disabled;

Do you mind adding this quick change?

StefanNieuwenhuis committed:
Update disabled button opacity

According to the designs, the disabled plans label should have an opacity of 0.5. I have updated the CSS to match this spec. I have created quite a specific selector since we do not want to affect the rest of the button styles (e.g. button opacity).

ciampo commented on packages/plans-grid/src/plans-table/style.scss:
```diff
@@ -228,6 +228,10 @@
 		color: var( --studio-gray-90 );
 		text-decoration: line-through;
 
+		span {
+			opacity: 0.5;
+		}
+
```
I know this may sound a bit of a nitpick, but I think we should apply those styles to the whole `.components-button` button — thus overriding the default `opacity: 0.3`

It looks like we could just move the

```scss
&:disabled {
  opacity: 0.5;
}
```

rule from being specific for the primary button, to instead apply to all `.plan-item__select-button.components-button` buttons — what do you think?

StefanNieuwenhuis replied:
I agree that remaining consistency is the best option but that's where it gets confusing as well. According to the design, the border opacity should be 0.64...

<img width="664" alt="Screenshot 2020-11-30 at 13 49 27" src="https://user-images.githubusercontent.com/12104589/100612296-0cf96c00-3313-11eb-9eb8-e456704c87b6.png">

I'm all in favor of applying `opacity: 0.5;` to both border & label. What should we do?

ciampo replied:
Let's apply `opacity: 0.5` to the whole `button` for now. We can always add a custom `border-color` later on

StefanNieuwenhuis committed:
Update opacity for disabled plan button

StefanNieuwenhuis committed:
Update opacity for primary plan button

StefanNieuwenhuis committed:
Update css selectors

The primary button disabled state was compromised by earlier changes. This commit fixes that by excluding the .is-primary selector.

ciampo reviewed (approved):
LGTM. Thank you for iterating over those changes!
